From: Joan Torres Lopez <joantolo@redhat.com>
Date: Tue, 27 Jan 2026 13:33:36 +0100
Subject: Revert "main: Register session with GDM on startup"

This depends on GDM changes that are not in this release.

This reverts commit 6457a1454a05bac8f7e172c17fbcb2b5baeae619.

Origin: upstream, 49.4, commit:ba60daead6a54ec02ed5aa2841e72cd5acb591cb
Bug: https://gitlab.gnome.org/GNOME/gnome-shell/-/issues/8983
Bug: https://gitlab.gnome.org/GNOME/mutter/-/issues/4572
Bug-Debian: https://bugs.debian.org/1125273
Bug-Debian: https://bugs.debian.org/1125275
Bug-Debian: https://bugs.debian.org/1125941
Bug-Debian: https://bugs.debian.org/1126174
---
 js/ui/main.js | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/js/ui/main.js b/js/ui/main.js
index 90e78a1..c911ab2 100644
--- a/js/ui/main.js
+++ b/js/ui/main.js
@@ -323,8 +323,6 @@ async function _initializeUI() {
     extensionManager = new ExtensionSystem.ExtensionManager();
     extensionManager.init();
 
-    LoginManager.registerSessionWithGDM();
-
     if (sessionMode.isGreeter && screenShield) {
         layoutManager.connect('startup-prepared', () => {
             screenShield.showDialog();
@@ -371,6 +369,8 @@ async function _initializeUI() {
             sessionMode.currentMode !== 'initial-setup')
             _handleLockScreenWarning();
 
+        LoginManager.registerSessionWithGDM();
+
         if (perfModule) {
             const perfOutput = GLib.getenv('SHELL_PERF_OUTPUT');
             Scripting.runPerfScript(perfModule, perfOutput);
